### PR TITLE
interop: devnet InteropDeployer script

### LINKED ISSUE #10608: Modularize ops-bedrock devnet
**Is your feature request related to a problem? Please describe.**

The current ops-bedrock devnet is defined in one docker-compose.yml file. This makes spinning up multiple L2s on devnet difficult. We need to have arbitrary number of L2s defined in devnet to test interoperability of opstack chains. 

**Describe the solution you'd like**

Split the docker compose yaml file into `base.yml` defining containers for L1 layer, and `overlays-l2.yml` defining containers for L2 layer. Then we can do `docker compose -f base.yml -f overlays-l2-A.yml -f overlays-l2-B ...` The role and responsibility of the [devnet launch script](https://github.com/ethereum-optimism/optimism/blob/develop/bedrock-devnet/devnet/__init__.py) is to manage the meta data files into separate directories so that user can deploy arbitrarily number of L2s to the same devnet. 

In order to deduplicate the container definitions in `overlays-l2-x.yml`, we use [extends](https://docs.docker.com/compose/multiple-compose-files/extends/) directive to inherit common configuration for any L2 layer, overriding chain specific metadata and configuration files for each L2 deployments

**Describe alternatives you've considered**

**Defining additional L2 chains statically in docker compose file**: It loses extensibility to multiple L2 chains if we are going to test interop message across arbitrary number of chains. If testing interoperability of two chains is enough, this could also be a reasonable choice since it incurs lesser change to current codebase

**Additional context**

### PR DESCRIPTION
**Pull request by jinmel**
- Deploy script for multiple L2 chains in devnet
 - resolves #10608

**Description**

The interop deploy script aims to enable modular deployments of L2 possible. As stated in  #10608, this PR achieves:

- per-chain deploy configuration by each devnet
- per-chain l1-allocs but unified L1 genesis 

**Tests**

Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.

**Additional context**

Since there are many tech debts in the deploy script from the CI/CD, e2e integration test dependencies, interop deploy script extends the current deploy script, where the interop chain shares the superchain configuration.  

**Metadata**

- Fixes #[Link to Issue]

### CODE REVIEW

**Comment by tynes** [comment]
Appreciate the PR @jinmel! It is going to be difficult to maintain this diff as its a big copy/paste of a lot of code, we can reduce the diff with the following pattern:

```solidity
contract InteropDeployer {

  function run() public {
    uint256 id = vm.snapshot();
    vm.setEnv("", vm.envOr("", "")); // for each config value
    Deployer.setUp();

    Deployer.runWithStateDump();

    vm.revert(id);

    vm.setEnv("", vm.envOr("", "")); // for each config value
    Deployer.runWithStateDump();

    vm.loadAllocs("path-to-first-output.json");
    vm.loadAllocs("path-to-second-output.json");

    vm.dumpState("final-output.json");
  }
}
```

This pseudocode isn't exhaustive, but we should be able to use the deploy script directly instead of making a new script with the same functionality

**Comment by jinmel** [comment]
@tynes that is really nice set of vm funcs i didn't know. thanks for your advice

File: bedrock-devnet/devnet/__init__.py [hunk 1]
```diff
@@ -300,6 +360,83 @@ def devnet_deploy(paths):
...
+    log.info(f'Using batch inbox {batch_inbox_address}')
+
+    # Set up the base docker environment.
+    docker_env = {
+        'PWD': paths.ops_bedrock_dir,
+    }
+
+    # Selectively set the L2OO_ADDRESS or DGF_ADDRESS if using L2OO.
+    # Must be done selectively because op-proposer throws if both are set.
+    if DEVNET_L2OO:
```
Comment (tynes) [inline bedrock-devnet/devnet/__init__.py]: We could simplify here and force fault proofs and no plasma

Comment (jinmel) [reply bedrock-devnet/devnet/__init__.py]: done

**Comment by jinmel** [comment]
@tynes I am seeing the implementation of L1 contracts being equal across rollups, and their proxies are different. Is this intented?

**Comment by tynes** [comment]
Yes this is intended! It was designed this way, to enable many chains to use the same implementations to make upgrades cheaper

**Comment by tynes** [comment]
Need to try this out locally before approving but super appreciate the help here! Is it working locally for you?

**Comment by jinmel** [comment]
@tynes I am currently debugging why op-batcher is not working for first rollup. It exits immediately after printing execution reverted, which seems to be an issue related to some underlying contracts. I have confirmed that the DisputeGameFactoryProxy addresses are different for each rollups, so it must be either I messed up somewhere in other configurations.

**Pull request closed without merging**